boxoffice/extapi/stats.py

```diff
@@ -0,0 +1,26 @@
+from boxoffice import db, app
+from boxoffice.models import ItemCollection, LineItem, Order, ORDER_STATUS
+from datetime import datetime as dt
+import requests
+from urllib import urlencode
+import json
+from tabulate import tabulate
+from rq.decorators import job
+
+@job('boxoffice')
+def post_stats(id, order_type):
+    if id is None:
+        return None
+
+    with app.test_request_context():
+        item_collection =  ItemCollection.query.get(id)
+        started = [ORDER_STATUS.PURCHASE_ORDER]
+        sold = [ORDER_STATUS.SALES_ORDER, ORDER_STATUS.INVOICE]
+        cancelled = [ORDER_STATUS.CANCELLED]
+        results = []
+        for item in item_collection.items:
+            results.append([item.title, LineItem.query.join(Order).filter(LineItem.item == item, Order.status.in_(started)).count(), LineItem.query.join(Order).filter(LineItem.item == item, Order.status.in_(sold)).count(), LineItem.query.join(Order).filter(LineItem.item == item, Order.status.in_(cancelled)).count()])
+        # Prepend and append tilde for the stats to be published with raw text highlighting
+        stats = ":moneybag: " + order_type + ": ```" + tabulate(results, headers = ["Event", "Initiated", "Sold", "Cancelled"]) + "```"
+        data={"username": "ka-ching", "text": stats}
+        response = requests.post(app.config['WEBHOOK_URL'], data = json.dumps(data))
```

Comment:
The `WEBHOOK_URL` is going to be specific to an organization, right? So, it makes sense for it to be read from an organization's `details` field.